app/controllers/concerns/unconfirmed_user_concern.rb

```diff
@@ -8,6 +8,42 @@ def with_unconfirmed_user
     yield if block_given?
   end
 
+  def validate_token
+    with_unconfirmed_user do
+      result = EmailConfirmationTokenValidator.new(@user).submit
+
+      analytics.track_event(Analytics::EMAIL_CONFIRMATION, result.to_h)
+
+      if result.success?
+        process_successful_confirmation
+      else
+        process_unsuccessful_confirmation
+      end
+    end
+  end
+
+  def process_successful_confirmation
+    if !@user.confirmed?
```

Comment:
codeclimate says this is un-tested?

Reply:
This should be removed from the concern since it's defined in each controller. I will fix.